index.go

```diff
@@ -126,6 +127,10 @@ func addDocumentR(bucket *bbolt.Bucket, parts []FieldIndexer, cKey Key, ref Refe
 			key := ComposeKey(cKey, m)
 			_ = addRefToBucket(bucket, key, ref)
 		}
+		if len(matches) == 0 {
```

Comment:
I dont understand this "if"?

Reply:
This means that the JSON path query for this index part didn't return any result => a nil value. A nil value is reflected in the compound key with two adjacent separators:

for an index with 3 indexed fields, where the third part resulted in a nil value
```key = value1 + 0x10 + value2 + 0x10 + 0x10```